packages/web-components/src/components/va-date/va-date.tsx

```diff
@@ -0,0 +1,208 @@
+   * This prop should be leveraged to display any custom validations needed for this component
+   */
+  @Prop() error: string;
+
+  /**
+   * Set the `min` value on the year input.
+   */
+  @Prop() minYear: number;
+
+  /**
+   * Set the `max` value on the year input.
+   */
+  @Prop() maxYear: number;
+
+  /**
+   * Set the default date value must be in YYYY-MM-DD format.
+   */
+  @Prop({ mutable: true }) value: string;
+
+  /**
+   * Fires when the date input loses focus after its value was changed
+   */
+  @Event({
+    composed: true,
+    bubbles: true,
+  })
+  dateChange: EventEmitter;
+
+  /**
+   * Fires when the date input loses focus
+   */
+  @Event({
+    composed: true,
+    bubbles: true,
+  })
+  dateBlur: EventEmitter;
+
+  private handleDateBlur = (event: FocusEvent) => {
+    this.dateBlur.emit(event);
+  };
+
+  private handleDateChange = (event: Event) => {
+    const target = event.target as HTMLSelectElement | HTMLInputElement;
+    let [currentYear, currentMonth, currentDay] = (this.value || '').split('-');
+    if (target.classList.contains('select-month')) {
+      currentMonth = target.value;
+    }
+    if (target.classList.contains('select-day')) {
+      currentDay = target.value;
+    }
+    if (target.classList.contains('input-year')) {
+      currentYear = target.value;
+    }
+    // Use a leading zero for numbers < 10
+    const numFormatter = new Intl.NumberFormat('en-US', {
+      minimumIntegerDigits: 2,
+    });
+
+    // If Month or Date Select are chosen as empty return ''
+    // Otherwise convert numbers less than 10 to have a leading 0
+    this.value = `${currentYear}-${
+      currentMonth ? numFormatter.format(parseInt(currentMonth)) : ''
+    }-${currentDay ? numFormatter.format(parseInt(currentDay)) : ''}`;
+
+    // This event should always fire to allow for validation handling
+    this.dateChange.emit(event);
+  };
+
+  /**
+   * Whether or not an analytics event will be fired.
+   */
+  @Prop() enableAnalytics: boolean;
+
+  /**
+   * The event used to track usage of the component. This is emitted when an
+   * input value changes and enableAnalytics is true.
+   */
+  @Event({
+    eventName: 'component-library-analytics',
+    composed: true,
+    bubbles: true,
+  })
+  componentLibraryAnalytics: EventEmitter;
+
+  render() {
+    const {
+      required,
+      label,
+      name,
+      error,
+      maxYear,
+      minYear,
+      handleDateBlur,
+      handleDateChange,
+      value,
+    } = this;
+
+    const [year, month, day] = (value || '').split('-');
+
+    // Convert string to number to remove leading 0 when matching to days object
+    const daysForSelectedMonth =
+      parseInt(month, 10) > 0 ? days[parseInt(month, 10)] : [];
+
+    // Check validity of date if invalid provide message and error state styling
+    const dateInvalid =
+      required && (!isFullDate(value) || day > daysForSelectedMonth.length)
+        ? 'Please provide a valid date'
+        : null;
```

Comment:
Thank you for adding this in. Could you please add a test for it?

Reply:
@k80bowman yes testing suite has been updated